clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerAuthenticator.java

```diff
@@ -381,9 +381,11 @@ private void handleSaslToken(byte[] clientToken) throws IOException {
                 // For versions with SASL_AUTHENTICATE header, send a response to SASL_AUTHENTICATE request even if token is empty.
                 ByteBuffer responseBuf = responseToken == null ? EMPTY_BUFFER : ByteBuffer.wrap(responseToken);
                 sendKafkaResponse(requestContext, new SaslAuthenticateResponse(Errors.NONE, null, responseBuf));
-            } catch (SaslException e) {
+            } catch (Exception e) {
```

Comment:
Is it intentional that we catch all exceptions, not just SaslException and AuthenticationException?

Reply:
No, fixed.